pkg/controller/snatpolicy/snatpolicy_controller.go

```diff
@@ -149,6 +150,7 @@ func (r *ReconcileSnatPolicy) addFinalizer(m *aciv1.SnatPolicy) error {
 // Cleanup steps to be done when snatPolicy resource is getting deleted.
 func (r *ReconcileSnatPolicy) finalizeSnatPolicy(reqLogger logr.Logger, m *aciv1.SnatPolicy) error {
 	if len(m.Status.SnatPortsAllocated) != 0 {
+		time.Sleep(time.Second * 5)
```

Comment:
Stating the obvious, this seems flaky. Is there any other way to do it? What if the system is more loaded and 5 sec is not enough?